src/js/AuditRule.js

```diff
@@ -124,12 +124,20 @@ axs.AuditRule.prototype.addElement = function(elements, element) {
  * @param {Node} node
  * @param {function(Element): boolean} matcher
  * @param {Array.<Element>} collection
- * @param {ShadowRoot=} opt_shadowRoot The nearest ShadowRoot ancestor, if any.
+ * @param {Object=} options: optional parameters: shadowRoot = The nearest ShadowRoot ancestor, if any. ignoredSelectors = The array of selectors to ignore, if any.
```

Comment:
I'm finding conflating the optional opt_shadowRoot argument with options confusing. How about making this into two methods, along these lines:

/**
 * @param {Node} node
 * @param {function(Element): boolean} matcher
 * @param {Array<string>=} opt_ignoreSelectors
 */
axs.AuditRule.collectMatchingElements = function(node, matcher, opt_ignoreSelectors) {
    axs.AuditRule.collectMatchingElementsInternal(node, matcher, null, opt_ignoreSelectors);
};

/**
 * @param {Node} node
 * @param {function(Element): boolean}
 * @param {?ShadowRoot} shadowRoot The nearest ShadowRoot ancestor, if any
 * @param {Array<string=} opt_ignoreSelectors
 */
axs.AuditRule.collectMatchingElementsInternal = function(node, matcher, shadowRoot, opt_ignoreSelectors) {
...
}